kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
Ah get returns null. Perhaps we should refactor that so `get` calls `getMandatory` instead and `get` returns null if it's a 404? That way we can use the proper error from the `GET` rather than this synthetic error.